DPG/Tasks/AOTTrack/tagAndProbeDmesons.cxx

```diff
@@ -1393,6 +1399,57 @@ struct ProbeThirdTrack {
 
   void processDummy(aod::Collisions const&) {}
   PROCESS_SWITCH(ProbeThirdTrack, processDummy, "Dummy process function that does nothing", false);
+
+  void processGeneratedDstarToDzeroPi(aod::McCollision const& mcCollision, aod::McParticles const& mcParticles)
+  {
+
+    if (std::abs(mcCollision.posZ()) > 10.)
+      return;
+    std::array<int, 3> arrDstar = {211, -321, 211};
+    int8_t* sign = nullptr;
+    std::vector<int> listIndexDaughters;
+    float ptDzero = -1, yDzero = -999, ptSoftPion = -1, etaSoftPion = -999, ptminTagDaughers = 9999., etamaxTagDaugthers = 0.;
+    int indexProbe;
+    for (auto& mcpart : mcParticles) {
+      //  Printf("particle id: %d",mcpart.pdgCode());
+      if (RecoDecay::isMatchedMCGen<true, true, 3>(mcParticles, mcpart, constants::physics::Pdg::kDStar, arrDstar, true, sign, -1, &listIndexDaughters)) {
```

Comment:
Two steps are sufficient for D* -> D0pi -> Kpipi
```suggestion
      if (RecoDecay::isMatchedMCGen<true, true, 3>(mcParticles, mcpart, constants::physics::Pdg::kDStar, arrDstar, true, sign, 2, &listIndexDaughters)) {
```

Reply:
ok, but I think that it does not change anything, since the step number is checked to decide whether to break the loop.